basecoin/store/src/impls/in_memory.rs

```diff
@@ -1,18 +1,59 @@
+use core::convert::Infallible;
+
 use ics23::CommitmentProof;
 use tendermint::hash::Algorithm;
 use tendermint::Hash;
 use tracing::trace;
 
 use crate::avl::{AsBytes, AvlTree};
 use crate::context::{ProvableStore, Store};
-use crate::types::{Height, Path, State};
+use crate::types::{Height, Path, RawHeight, State};
+
+#[derive(Debug, Clone, Default)]
+pub struct PrunedVec<T> {
+    vec: Vec<T>,
+    pruned: usize,
+}
+
+impl<T> PrunedVec<T> {
+    pub fn push(&mut self, value: T) {
+        self.vec.push(value);
+    }
+
+    pub fn get(&self, index: usize) -> Option<&T> {
+        self.vec.get(index.checked_sub(self.pruned)?)
+    }
+
+    pub fn last(&self) -> Option<&T> {
+        self.vec.last()
+    }
+
+    pub fn len(&self) -> usize {
```

Comment:
Why does the `len` method return the total number of elements, including the ones that were pruned? I would expect it to only return the current number of elements that are accessible.

Reply:
02df4a3